prov/sockets/src/sock_dom.c

```diff
@@ -331,11 +334,11 @@ static int sock_regattr(struct fid *fid, const struct fi_mr_attr *attr,
 	key = (dom->attr.mr_mode == FI_MR_BASIC) ?
 		sock_get_mr_key(dom) : attr->requested_key;
 
-	res = rbtInsert(dom->mr_heap, (void *)key, _mr);
+	_mr->key = key;
+	res = rbtInsert(dom->mr_heap, (void *)&_mr->key, _mr);
```

Comment:
Please drop casts in above code that aren't needed.  E.g. mr_key is a void *.  And it is usually safe to cast to/from void * and another pointer.